rotkehlchen/api/rest.py

```diff
@@ -4197,3 +4201,61 @@ def get_binance_savings_history(
             query_filter=query_filter,
             value_filter=value_filter,
         )
+
+    def get_ens_avatar(self, ens_name: str, match_header: Optional[str]) -> Response:
+        """
+        Searches for ENS avatar of the given `ens_name`.
+        If found returns the avatar, otherwise returns 404 status code.
+
+        Also supports etag mechanism that helps with caching on the API caller side.
+        """
+        avatar_path = self.rotkehlchen.data_dir / 'avatars' / f'{ens_name}.png'
+        md5_hash = None
+        if avatar_path.exists():
+            md5_hash = file_md5(avatar_path)
+
+        if md5_hash and match_header and match_header == md5_hash:
+            # Response content unmodified
+            return make_response(
+                (
+                    b'',
+                    HTTPStatus.NOT_MODIFIED,
+                    {'mimetype': 'image/png', 'Content-Type': 'image/png'},
+                ),
+            )
+
+        dbens = DBEns(self.rotkehlchen.data.db)
+        try:
+            last_update = dbens.get_last_avatar_update(ens_name)
+        except InputError as e:
+            return api_response(
+                result={'result': None, 'message': str(e)},
+                status_code=HTTPStatus.CONFLICT,
+            )
+        if last_update is None or ts_now() - last_update > ENS_AVATARS_REFRESH:
+            # If avatar for this ens name has never been checked or the avatar has expired
+            # then we try to download.
+            try_download_ens_avatar(
+                eth_inquirer=self.rotkehlchen.chains_aggregator.ethereum.node_inquirer,
+                avatars_dir=Path(self.rotkehlchen.data_dir) / 'avatars',
+                ens_name=ens_name,
+            )
+
+        if avatar_path.exists() is False:  # No avatar found, return 404
```

Comment:
nitpick: maybe use `.is_file()` instead
```suggestion
        if avatar_path.is_file() is False:  # No avatar found, return 404
```